ddtrace/tracer/civisibility_writer.go

```diff
@@ -0,0 +1,119 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package tracer
+
+import (
+	"sync"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// Constants defining the payload size limits for agentless mode.
+const (
+	// agentlessPayloadMaxLimit is the maximum payload size allowed, indicating the
+	// maximum size of the package that the intake can receive.
+	agentlessPayloadMaxLimit = 5 * 1024 * 1024 // 5 MB
```

Comment:
Quick question, what is this number based on? (Asking because trace size is bigger)

Reply:
It's a limitation from the Event Platform Intake, the max allowed payload size is 5MB there

Reply:
Gotcha, thanks